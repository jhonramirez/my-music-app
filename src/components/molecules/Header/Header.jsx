import { useNavigate } from "react-router-dom";
import { useSelector } from "react-redux";
import { Navlink, Label, Button } from "../..";
import { useInfoUser } from "../../../hooks";
import { icoLogout } from "../../../svg";
import "./Header.scss";

export const Header = () => {
  const { user } = useSelector((state) => state.auth);
  const navigate = useNavigate();
  const token = sessionStorage.getItem("token");
  useInfoUser(token);

  const onLogout = () => {
    sessionStorage.removeItem("token");
    navigate("/login", {
      replace: true,
    });
  };

  return (
    <div className="header">
      <div className="header__left">
        <Navlink
          navigate="/"
          img="Mymusicapp.png"
          alt="My music App"
          className="header__imghome"
        />
        <Navlink
          navigate="/favorites"
          name="Favoritos"
          className="header__favorite"
        />
      </div>
      <div className="header__right">
        <form onSubmit={onLogout}>
          <Label name={user?.display_name} className="header__name" />
          <Button label={icoLogout} className="header__logout" />
        </form>
      </div>
    </div>
  );
};
