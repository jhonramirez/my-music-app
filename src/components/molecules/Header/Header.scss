.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 2px 18px;
  background-color: rgb(17, 17, 16);
  text-decoration: none;
  position: fixed;
  width: 99%;
  z-index: 1; 
  &__left{
    display: flex;
    text-align: center;
    align-items: center;
  }
  &__right{
    display: flex;
    text-align: center;
    align-items: center;
  }
  &__favorite {
    color: #dad4ce;
    text-decoration: none;
  }
  &__name {
    color: #ffff;
    padding: 10px;
  }
  &__imghome {
    width: 25px;
    height: 25px;
  }
  &__logout{
    padding-right: 10px;
  }
  &__favorite:hover {
    color: rgb(240, 136, 39);
  }
}
