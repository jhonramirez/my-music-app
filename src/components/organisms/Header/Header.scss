.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 2px 18px;
  background-color: rgb(17, 17, 16);
  text-decoration: none;
  position: fixed;
  width: 99%;
  z-index: 1; 
  &__favorite {
    color: #dad4ce;
    text-decoration: none;
  }
  &__name {
    color: #ffff;
    padding: 10px;
  }
  &__imghome {
    width: 25px;
    height: 25px;
  }
  &__logout{
    padding-right: 10px;
  }
}

.header__favorite:hover {
  color: rgb(240, 136, 39);
}