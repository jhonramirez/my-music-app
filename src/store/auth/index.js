export * from "./authSlice";
export * from "./playlistSlice";