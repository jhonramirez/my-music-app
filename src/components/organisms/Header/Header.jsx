import { useNavigate } from "react-router-dom";
import { useSelector } from "react-redux";
import { Navlink, Label, Button } from "../..";
import { useInfoUser } from "../../../hooks";
import { icoLogout } from "../../../svg";
import "./Header.scss";

export const Header = () => {
  const { user } = useSelector((state) => state.auth);
  const navigate = useNavigate();
  const token = sessionStorage.getItem("token");
  useInfoUser(token);

  const onLogout = () => {
    sessionStorage.removeItem("token");
    navigate("/login", {
      replace: true,
    });
  };

  return (
    <div className="header">
        <Navlink
          navigate="/"
          img="Mymusicapp.png"
          alt="My music App"
          className="header__imghome"
        />
        <Navlink navigate="/favorites" name="Favoritos" className="header__favorite" />

        <form onSubmit={onLogout}>
          <Label name={user?.display_name} className="header__name" />
          <Button onClick={() => onLogout()} icon= {icoLogout} className="header__logout" />
        </form>
    </div>
  );
};
