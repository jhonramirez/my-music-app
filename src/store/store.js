import { configureStore } from "@reduxjs/toolkit";
import { authSlice, playlistSlice } from "./auth";

export const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    playlist: playlistSlice.reducer,
  },
});
