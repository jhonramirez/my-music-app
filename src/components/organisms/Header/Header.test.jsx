import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import { Header } from "./Header";
import { configureStore } from "@reduxjs/toolkit";
import { authSlice } from "../../../store";

const mockedUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUseNavigate,
}));

const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
  },
  preloadedState: {
    auth: {
      user: { display_name: "proof" },
    },
  },
});

describe("Pruebas en <Header/>", () => {
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test("Debe tener un texto 'Favoritos'", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    );
    const title = screen.getByText("Favoritos");
    expect(title).toBeInTheDocument();
  });

  test("Debe tener la imagen 'Mymusicapp.png'", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    );
    const img = screen.getAllByRole("img")[0];
    expect(img.src).toContain("Mymusicapp.png");
  });

  test("Se debe llamar el onLogout y navigate cuando se hace click en el botÃ³n logout", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    );
    const onLogoutElement = screen.getByRole("button");
    fireEvent.click(onLogoutElement);
    expect(mockedUseNavigate).toHaveBeenCalledWith("/login", { replace: true });
  });

  test("Debe existir texto proof", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    );
    const userName = screen.getByText("proof");
    expect(userName).toBeDefined();
  });
});
