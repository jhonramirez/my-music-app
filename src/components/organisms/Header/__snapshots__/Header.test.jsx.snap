// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Header/> Debe hacer match con el snapshot 1`] = `
<div>
  <section
    class="header"
  >
    <article
      class="header__left"
    >
      <img
        alt="My music App"
        class="header__imghome"
        src="Mymusicapp.png"
      />
      <a
        aria-current="page"
        class="header__home active"
        href="/"
      >
        Inicio
      </a>
      <a
        class="header__favorite"
        href="/favorites"
      >
        Favoritos
      </a>
    </article>
    <form
      class="header__right"
    >
      <label
        class="header__name"
        data-testid="label"
      >
        proof
      </label>
      <button
        class="btn btn__black"
        data-testid="button"
      >
        <svg
          class="feather feather-log-out"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
          />
          <polyline
            points="16 17 21 12 16 7"
          />
          <line
            x1="21"
            x2="9"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
    </form>
  </section>
</div>
`;
